### PR TITLE
Refactor: Project Overview fetching

### PR DESCRIPTION
**Pull request by Tymek**
## About the changes
https://linear.app/unleash/issue/1-616/refactor-project-endpoint

### CODE REVIEW

File: frontend/src/hooks/useLoading.ts [hunk 1]
```diff
@@ -2,11 +2,13 @@ import { createRef, useLayoutEffect } from 'react';
 
 type refElement = HTMLDivElement;
 
-const useLoading = (loading: boolean) => {
+const useLoading = (loading: boolean, label = 'true') => {
```
Comment (kwasniew) [inline frontend/src/hooks/useLoading.ts:5]: IMO this API is too specific. As a user of useLoading I'd expect (loading: boolean, selector = '[data-loading]') that I can override. label that goes into data-loading selector is not very intuitive from the user's perspective.

File: frontend/src/component/project/Project/ProjectInfo/ProjectInfo.tsx [hunk 2]
```diff
@@ -50,8 +51,11 @@ const ProjectInfo = ({
     const { uiConfig, isEnterprise } = useUiConfig();
     const theme = useTheme();
     const isSmallScreen = useMediaQuery(theme.breakpoints.down('md'));
+    const hasChangeRequestsEnabled = useChangeRequestConfig(id).data?.some(
```
Comment (kwasniew) [inline frontend/src/component/project/Project/ProjectInfo/ProjectInfo.tsx:54]: There's a hook useChangeRequestsEnabled.ts that returns isChangeRequestConfiguredInAnyEnv

File: frontend/src/component/project/Project/ProjectInfo/ChangeRequestsWidget.tsx [hunk 3]
```diff
@@ -11,6 +11,8 @@ import {
 import { useProjectChangeRequests } from 'hooks/api/getters/useProjectChangeRequests/useProjectChangeRequests';
 import { WidgetFooterLink } from './WidgetFooterLink';
 
+const LOADING_LABEL = 'change-requests-widget';
```
Comment (kwasniew) [inline frontend/src/component/project/Project/ProjectInfo/ChangeRequestsWidget.tsx:14]: why do we need custom selector here?

**Review by kwasniew — CHANGES_REQUESTED** [review]

File: frontend/src/component/project/Project/ProjectInfo/ChangeRequestsWidget.tsx (hunk 3 above)
Comment (Tymek) [reply frontend/src/component/project/Project/ProjectInfo/ChangeRequestsWidget.tsx:14]: Data for change requests is being loaded from another endpoint (that isn't available in unleash-core)

**Review by kwasniew — APPROVED** [review]

**Pull request merged**